src/javaharness/java/arcs/api/PECInnerPortImpl.java

```diff
@@ -155,6 +160,60 @@ public void SynchronizeProxy(StorageProxy storageProxy, Consumer<PortableJson> c
         postMessage(message);
     }
 
+    @Override
+    public void HandleStore(StorageProxy storageProxy, Consumer<PortableJson> callback, PortableJson data, String particleId) {
+        PortableJson message = jsonParser.parse("{}");
+        message.put(MESSAGE_TYPE_FIELD, HANDLE_STORE_MSG);
+        PortableJson body = jsonParser.parse("{}");
+        body.put(PROXY_HANDLE_ID_FIELD, mapper.identifierForThing(new Thing<StorageProxy>(storageProxy)));
+        body.put(PROXY_CALLBACK_FIELD, mapper.createMappingForThing(
+            new Thing<Consumer<PortableJson>>(callback), /* requestedId= */ null));
+        body.put(DATA_FIELD, data);
+        body.put(PARTICLE_ID_FIELD, particleId);
+        message.put(MESSAGE_BODY_FIELD, body);
+        postMessage(message);
+    }
+
+    @Override
+    public void HandleToList(StorageProxy storageProxy, Consumer<PortableJson> callback) {
+        PortableJson message = jsonParser.parse("{}");
+        message.put(MESSAGE_TYPE_FIELD, HANDLE_TO_LIST_MSG);
+        PortableJson body = jsonParser.parse("{}");
+        body.put(PROXY_HANDLE_ID_FIELD, mapper.identifierForThing(new Thing<StorageProxy>(storageProxy)));
+        body.put(PROXY_CALLBACK_FIELD, mapper.createMappingForThing(
+            new Thing<Consumer<PortableJson>>(callback), /* requestedId= */ null));
+        message.put(MESSAGE_BODY_FIELD, body);
+        postMessage(message);
+    }
+
+    @Override
+    public void HandleRemove(StorageProxy storageProxy, Consumer<PortableJson> callback, PortableJson data, String particleId) {
+        PortableJson message = jsonParser.parse("{}");
```

Comment:
This method impl is identical to RemoveMultiple except for the message type, suggest adding helper they both use just to make the code duplication more apparent.

Reply:
thanks, done (plus some more refactoring)